validator/valnode/redis/consumer.go

```diff
@@ -113,14 +131,39 @@ func (s *ValidationServer) Start(ctx_in context.Context) {
 			case <-readyStreams:
 				log.Trace("At least one stream is ready")
 				return // Don't block Start if at least one of the stream is ready.
-			case <-time.After(s.streamTimeout):
+			case <-time.After(s.config.StreamTimeout):
 				log.Error("Waiting for redis streams timed out")
 			case <-ctx.Done():
 				log.Info("Context done while waiting redis streams to be ready, failed to start")
 				return
 			}
 		}
 	})
+	for i := 0; i < workers; i++ {
+		s.StopWaiter.LaunchThread(func(ctx context.Context) {
+			for {
+				var work workUnit
+				select {
+				case <-ctx.Done():
+					return
+				case work = <-workQueue:
+				}
+				valRun := s.spawner.Launch(work.req.Value, work.moduleRoot)
+				res, err := valRun.Await(ctx)
+				if err != nil {
+					log.Error("Error validating", "request value", work.req.Value, "error", err)
+				}
+				if err := s.consumers[work.moduleRoot].SetResult(ctx, work.req.ID, res); err != nil {
+					log.Error("Error setting result for request", "id", work.req.ID, "result", res, "error", err)
+				}
```

Comment:
Before, we do not `SetResult` if valRun returned an error but now we log an error and proceed to try and set this nil response in redis, wont this cause issues? As in wont we only want to set results of successful solved queries to redis?